dir.go

```diff
@@ -405,7 +403,7 @@ func (d *Directory) AddChild(name string, nd ipld.Node) error {
 
 // AddUnixFSChild adds a child to the inner UnixFS directory
 // and transitions to a HAMT implementation if needed.
-func (d *Directory) AddUnixFSChild(name string, node ipld.Node) error {
+func (d *Directory) AddUnixFSChild(c Child) error {
```

Comment:
As an internal helper type, `child` is fine (although I still feel it's unecessary). However, I really don't want to see it leak into an external interface. All it does is change calls to:

```go
d.AddUnixFSChild(name, node)
```

To:

```go
d.AddUnxiFSChild(mfs.Child{Name: name, Node: node})
```

Which is strictly more annoying to use and provides no additional safety, documentation, etc.

(sorry, I should have commented on this the first round but I assumed you'd replace `child` with the name/node in this case instead of exporting it)

Reply:
Thanks for pointing me to it, actually `AddUnixFSChild` shouldn't be exported in the first place, I'll also un-export child.